Mathlib/Analysis/NormedSpace/WeakOperatorTopology.lean

```diff
@@ -0,0 +1,276 @@
+  `A : E →WOT[𝕜] F` along filter `l` iff `y ((f a) x)` tends to `y (A x)` along the same filter.
+* `ContinuousLinearMap.toWOT`: the inclusion map from `E →L[𝕜] F` to the type copy
+* `ContinuousLinearMap.continuous_toWOT`: the inclusion map is continuous, i.e. the WOT is coarser
+  than the norm topology.
+* `ContinuousLinearMapWOT.withSeminorms`: the WOT is induced by the family of seminorms
+  `‖y (A x)‖` for `x : E` and `y : NormedSpace.Dual 𝕜 F`.
+
+## Notation
+
+* The type copy of `E →L[𝕜] F` endowed with the weak operator topology is denoted by
+  `E →WOT[𝕜] F`.
+* We locally use the notation `F⋆` for `NormedSpace.Dual 𝕜 F`.
+
+## Implementation notes
+
+In the literature, the WOT is only defined on maps between Banach spaces. Here, we generalize this
+a bit to `E →L[𝕜] F` where `F` is an normed space, and `E` actually only needs to be a vector
+space with some topology for most results in this file.
+-/
+
+open scoped Topology
+
+/-- The type copy of `E →L[𝕜] F` endowed with the weak operator topology, denoted as
+`E →WOT[𝕜] F`. -/
+@[irreducible]
+def ContinuousLinearMapWOT (𝕜 : Type*) (E : Type*) (F : Type*) [Semiring 𝕜] [AddCommGroup E]
+    [TopologicalSpace E] [Module 𝕜 E] [AddCommGroup F] [TopologicalSpace F] [Module 𝕜 F] := E →L[𝕜] F
+
+@[inherit_doc]
+notation:25 E " →WOT[" 𝕜 "]" F => ContinuousLinearMapWOT 𝕜 E F
+
+namespace ContinuousLinearMapWOT
+
+variable {𝕜 : Type*} {E : Type*} {F : Type*} [RCLike 𝕜] [AddCommGroup E] [TopologicalSpace E]
+  [Module 𝕜 E] [NormedAddCommGroup F] [NormedSpace 𝕜 F]
+
+local postfix:max "⋆" => NormedSpace.Dual 𝕜
+
+/-!
+### Basic properties common with `E →L[𝕜] F`
+
+The section copies basic non-topological properties of `E →L[𝕜] F` over to `E →WOT[𝕜] F`, such as
+the module structure, `FunLike`, etc.
+-/
+section Basic
+
+unseal ContinuousLinearMapWOT in
+instance instAddCommGroup : AddCommGroup (E →WOT[𝕜] F) :=
+  inferInstanceAs <| AddCommGroup (E →L[𝕜] F)
+
+unseal ContinuousLinearMapWOT in
+instance instModule : Module 𝕜 (E →WOT[𝕜] F) := inferInstanceAs <| Module 𝕜 (E →L[𝕜] F)
+
+variable (𝕜) (E) (F)
+
+unseal ContinuousLinearMapWOT in
+/-- The equivalence that sends a continuous linear map to the type copy endowed with the
+weak operator topology.  -/
+def _root_.ContinuousLinearMap.toWOT : (E →L[𝕜] F) ≃ₗ[𝕜] (E →WOT[𝕜] F) := LinearEquiv.refl 𝕜 _
+
+variable {𝕜} {E} {F}
+
+instance instFunLike : FunLike (E →WOT[𝕜] F) E F where
+  coe f :=  ((ContinuousLinearMap.toWOT 𝕜 E F).symm f : E → F)
+  coe_injective' := by intro; simp
+
+instance instContinuousLinearMapClass : ContinuousLinearMapClass (E →WOT[𝕜] F) 𝕜 E F where
+  map_add f x y := by simp only [DFunLike.coe]; simp
+  map_smulₛₗ f r x := by simp only [DFunLike.coe]; simp
+  map_continuous f := ContinuousLinearMap.continuous ((ContinuousLinearMap.toWOT 𝕜 E F).symm f)
+
+lemma _root_.ContinuousLinearMap.toWOT_apply {A : E →L[𝕜] F} {x : E} :
+    ((ContinuousLinearMap.toWOT 𝕜 E F) A) x = A x := rfl
```

Comment:
Are you thinking we don't want this to be `@[simp]` because it switches from WOT to CLM and we don't want to break the barrier? If so, you can just resolve this.

Reply:
Yeah I'm not too sure either whether we want this to be a simp lemma or not. Let's just leave it like this, we can always add the tag later if not having it turns out to be annoying.